src/js/core/methods/helpers/uploadFirmware.js

```diff
@@ -0,0 +1,36 @@
+/* @flow */
+
+import type { DefaultMessageResponse, MessageResponse } from '../../../device/DeviceCommands';
+import type { FirmwareRequest$ } from '../../../types/response';
+import * as trezor from '../../../types/trezor'; // flowtype only
+
+export const uploadFirmware = async (
+    typedCall: (type: string, resType: string, msg: Object) => Promise<DefaultMessageResponse>,
+    payload: Buffer,
+    offset?: number,
+    length?: number,
+    model: number,
+): trezor.Success => {
+    let response: MessageResponse<trezor.Success | FirmwareRequest$> = {};
+
+    if (model === 1) {
```

Comment:
i don't see this logic with strip 256 bytes for BL > 1.80? Shouldn't it be here?